libraries/AP_Compass/AP_Compass.h

```diff
@@ -480,7 +487,31 @@ friend class AP_Compass_Backend;
         // accumulated samples, protected by _sem, used by AP_Compass_Backend
         Vector3f accum;
         uint32_t accum_count;
-    } _state[COMPASS_MAX_INSTANCES];
+        // We only copy persistent params
+        void copy_from(const mag_state& state) {
```

Comment:
copy_from() is a bit long for the .h file, I'd move it to the .cpp

Reply:
will do